client/my-sites/theme/main.jsx

```diff
@@ -188,10 +193,51 @@ const BannerUpsellTitle = ( {
 	translate,
 	isSiteEligibleForManagedExternalThemes,
 	isMarketplaceThemeSubscribed,
+	isThemeAllowedOnSite,
+	themeTier,
 } ) => {
 	const bundleSettings = useBundleSettingsByTheme( themeId );
 	const isEnglishLocale = useIsEnglishLocale();
 
+	const premiumPlanTitle = () =>
+		isEnglishLocale ||
+		i18n.hasTranslation(
+			'Access this theme for FREE with a %(premiumPlanName)s or %(businessPlanName)s plan!'
+		)
+			? translate(
+					'Access this theme for FREE with a %(premiumPlanName)s or %(businessPlanName)s plan!',
+					{
+						args: {
+							premiumPlanName: getPlan( PLAN_PREMIUM ).getTitle(),
+							businessPlanName: getPlan( PLAN_BUSINESS ).getTitle(),
+						},
+					}
+			  )
+			: translate( 'Access this theme for FREE with a Premium or Business plan!' );
+
+	if ( ! isThemeAllowedOnSite ) {
+		switch ( THEME_TIERS[ themeTier.slug ].minimumUpsellPlan ) {
+			case PLAN_PERSONAL:
+				return isEnglishLocale ||
+					i18n.hasTranslation(
+						'Access this theme for FREE with a %(personalPlanName)s, %(premiumPlanName)s, or %(businessPlanName)s plan!'
+					)
+					? translate(
+							'Access this theme for FREE with a %(personalPlanName)s, %(premiumPlanName)s, or %(businessPlanName)s plan!',
+							{
+								args: {
+									personalPlanName: getPlan( PLAN_PERSONAL ).getTitle(),
+									premiumPlanName: getPlan( PLAN_PREMIUM ).getTitle(),
+									businessPlanName: getPlan( PLAN_BUSINESS ).getTitle(),
+								},
+							}
+					  )
+					: translate( 'Access this theme for FREE with a Personal, Premium, or Business plan!' );
+			case PLAN_PREMIUM:
+				return premiumPlanTitle();
+		}
+	}
```

Comment:
It would be great if we could unify all the translations in a single place so that we can avoid creating the same/similar translations over and over and over.

Reply:
The process that generates the translation file works by static analysis. The best you could have is a `i18n.preferFirstTranslation('String 1', 'String 2');`

Reply:
FWIW it looks like martech have been dropping the `hasTranslation` calls today e.g. in #85562 as it's better to have an untranslated message with the correct plan name.

Reply:
I don't think this is a reason to block merging since I'm sure there are plenty of other code places that are using `hasTranslation`. Changed it anyway.